### PR TITLE
GA Migration Python PostCommit Sharding Dataframe IO

### PR DESCRIPTION
**Pull request by MarcoRob**
As part of the migration of Precommit and Postcommit Jobs from Jenkins to GA in self-hosted runners, this PR contains:

- Migrated workflow [job-postcommit-python-dataframe-io.yml](https://github.com/apache/beam/blob/986596e0bb92ff0c1ab576723143c3ccf004e77b/.github/workflows/job-postcommit-python-dataframe-io.yml)

The migrated workflow was added to [CI.md](https://github.com/apache/beam/blob/986596e0bb92ff0c1ab576723143c3ccf004e77b/CI.md)

`DO NOT MERGE` until the effort to use self-hosted runners is completed https://github.com/apache/beam/pull/22703

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

File: .github/workflows/job-postcommit-python-dataframe-io.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        run: |
+          echo "PYTHON_VERSION=$(echo ${{ matrix.version }} | sed -e 's/\.//g')" >> $GITHUB_ENV
+      - name: Setup environment
+        uses: ./.github/actions/setup-self-hosted-action
+      - name: Run :sdks:python:test-suites:tox:${{ env.PYTHON_VERSION }}:testPy${{ env.PYTHON_VERSION }}Cloud
+        uses: ./.github/actions/gradle-command-self-hosted-action
+        with:
+          gradle-command: :sdks:python:test-suites:tox:py${{ env.PYTHON_VERSION }}:testPy${{ env.PYTHON_VERSION }}Cloud
+          arguments: "-Pposargs=${{env.tests}}"
+      - name: Run :sdks:python:test-suites:tox:y${{ env.PYTHON_VERSION }}:testPy${{ env.PYTHON_VERSION }}Cython
```
Comment (damccorm) [inline .github/workflows/job-postcommit-python-dataframe-io.yml:71]:
```suggestion
      - name: Run :sdks:python:test-suites:tox:py${{ env.PYTHON_VERSION }}:testPy${{ env.PYTHON_VERSION }}Cython
```

**Review by damccorm — APPROVED** [review]
LGTM (outside of the tiny nit), thanks!

File: .github/workflows/job-postcommit-python-dataframe-io.yml (hunk 1 above)
Comment (MarcoRob) [reply .github/workflows/job-postcommit-python-dataframe-io.yml]: Fix applied

File: .github/workflows/job-postcommit-python-dataframe-io.yml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+name: Python Postcommit Dataframe IO
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 */6 * * *'
+  push:
+    branches: ['master', 'release-*']
+    tags: ['v*']
+    paths: [ 'sdks/python/apache_beam/dataframe/io_it_test.py']
```
Comment (damccorm) [inline .github/workflows/job-postcommit-python-dataframe-io.yml:29]:
```suggestion
```

Postcommits should run no matter what

**Review by damccorm — APPROVED** [review]
LGTM, thanks

**Pull request closed without merging**